src/main/scala/com.snowplowanalytics.snowplow.micro/Configuration.scala

```diff
@@ -38,14 +38,29 @@ import java.nio.file.{Path, Paths}
 
 object Configuration {
 
+  sealed trait OutputFormat
+  object OutputFormat {
+    case object None extends OutputFormat
+    case object Tsv extends OutputFormat
+    case object Json extends OutputFormat
+  }
+
   object Cli {
-    final case class Config(collector: Option[Path], iglu: Option[Path], outputEnrichedTsv: Boolean)
+    final case class Config(collector: Option[Path], iglu: Option[Path], outputFormat: OutputFormat)
 
     private val collector = Opts.option[Path]("collector-config", "Path to HOCON configuration (optional)", "c", "config.hocon").orNone
     private val iglu = Opts.option[Path]("iglu", "Configuration file for Iglu Client", "i", "iglu.json").orNone
-    private val outputEnrichedTsv = Opts.flag("output-tsv", "Print events in TSV format to standard output", "t").orFalse
+    private val outputTsv = Opts.flag("output-tsv", "Print events in TSV format to standard output", "t").orFalse
+    private val outputJson = Opts.flag("output-json", "Print events in JSON format to standard output (with a separate key for each schema)").orFalse
+
+    private val outputFormat = (outputTsv, outputJson).mapN {
+      case (true, false) => OutputFormat.Tsv
+      case (false, true) => OutputFormat.Json
+      case (false, false) => OutputFormat.None
+      case (true, true) => throw new IllegalArgumentException("Cannot specify both --output-tsv and --json-output")
+    }
```

Comment:
By throwing an exception you are short-circuiting the argument parsing of `decline` and crashing the app brutally. That works but we avoid doing that in FP.

It seems that `decline` [doesn't support](https://ben.kirw.in/decline/usage.html#combining-options) exclusive arguments with an error message (we have to pick one by default).

It should be possible to write some custom behavior with [mapValidated](https://ben.kirw.in/decline/arguments.html#defining-your-own), something like this:

```suggestion
    private val outputFormat = (outputTsv, outputJson)
      .mapN { (_, _) }
      .mapValidated {
        case (true, false) => OutputFormat.Tsv.validNel[String]
        case (false, true) => OutputFormat.Json.validNel[String]
        case (false, false) => OutputFormat.None.validNel[String]
        case (true, true) => "Cannot specify both --output-tsv and --json-output".invalidNel[OutputFormat]
      }
```